File: base/src/main/groovy/com/knight/plugin/doublecheck/java/WeavedClass.java
package com.knight.plugin.doublecheck.java;


import java.io.Serializable;
import java.util.LinkedHashSet;
import java.util.Set;

public class WeavedClass implements Serializable {

    private String className;

    private Set<String> doubleCheckMethods = new LinkedHashSet<>();

    WeavedClass(String className) {
        this.className = className;
    }

    public void addDoubleCheckMethod(String methodSignature) {
        doubleCheckMethods.add(methodSignature);
    }

    public String getClassName() {
        return className;
    }

    public Set<String> getDoubleCheckMethods() {
        return doubleCheckMethods;
    }

    public boolean hasDoubleCheckMethod() {
        return doubleCheckMethods.size() > 0;
    }


}

File: knight-transform/src/main/kotlin/com/knight/transform/task/WeavedClass.kt
package transform.task

import java.io.Serializable
import java.util.LinkedHashSet

class WeavedClass(val className: String) : Serializable {

    val doubleCheckMethods = LinkedHashSet<String>()


    fun addDoubleCheckMethod(methodSignature: String) {
        doubleCheckMethods.add(methodSignature)
    }


    fun getDoubleCheckMethods(): Set<String> {
        return doubleCheckMethods
    }

    fun hasDoubleCheckMethod(): Boolean {
        return doubleCheckMethods.size > 0
    }

}